src/view.c

```diff
@@ -1530,32 +1567,28 @@ view_get_adjacent_output(struct view *view, enum view_edge edge)
 		return NULL;
 	}
 
+	struct wlr_box box = output_usable_area_in_layout_coords(output);
+	int lx = box.x + box.width / 2;
+	int ly = box.y + box.height / 2;
+
 	/* Determine any adjacent output in the appropriate direction */
 	struct wlr_output *new_output = NULL;
 	struct wlr_output *current_output = output->wlr_output;
 	struct wlr_output_layout *layout = view->server->output_layout;
-	switch (edge) {
-	case VIEW_EDGE_LEFT:
-		new_output = wlr_output_layout_adjacent_output(
-			layout, WLR_DIRECTION_LEFT, current_output, 1, 0);
-		break;
-	case VIEW_EDGE_RIGHT:
-		new_output = wlr_output_layout_adjacent_output(
-			layout, WLR_DIRECTION_RIGHT, current_output, 1, 0);
-		break;
-	case VIEW_EDGE_UP:
-		new_output = wlr_output_layout_adjacent_output(
-			layout, WLR_DIRECTION_UP, current_output, 0, 1);
-		break;
-	case VIEW_EDGE_DOWN:
-		new_output = wlr_output_layout_adjacent_output(
-			layout, WLR_DIRECTION_DOWN, current_output, 0, 1);
-		break;
-	default:
-		break;
+	enum wlr_direction direction = get_wlr_direction(edge);
+	new_output = wlr_output_layout_adjacent_output(layout, direction,
+		current_output, lx, ly);
+
+	/* Try wrapping */
+	if (!new_output) {
+		new_output = wlr_output_layout_farthest_output(layout,
+			opposite_direction(direction), current_output, lx, ly);
 	}
```

Comment:
Should we unconditionally do this? I think it should be a disabled-by-default behavior.